contributors/devel/api-conventions.md

```diff
@@ -620,18 +630,82 @@ containers:
 Delete operation will delete the first entry in the list that match the merge key.
 But there is validation to make sure no 2 entries with the same merge key will get in the list.
 
+Here is an example:
+
+Suppose we define a list of environment variables and we call them
+the original environment variables:
+```yaml
+env:
+  - name: ENV1
+    value: foo
+  - name: ENV2
+    value: bar
+  - name: ENV3
+    value: baz
+```
+
+Then the server appends another environment variable:
+```yaml
+env:
+  - name: ENV1
+    value: foo
+  - name: ENV2
+    value: bar
+  - name: ENV3
+    value: baz
+  - name: ENV4
+    value: server-added
+```
+
+Then the user wants to change it from the original to the following using `kubectl apply`:
+```yaml
+env:
+  - name: ENV1
+    value: foo
+  - name: ENV2
+    value: bar
+  - name: ENV5
+    value: new-env
+```
+
+The patch will looks like:
+```yaml
+$preserveOrderList/env:
+  - name: ENV1
+  - name: ENV2
+  - name: ENV5
+env:
+  - name: ENV3
+    $patch: delete
+  - name: ENV5
+    value: new-env
+```
+
+After server applying the patch:
+```yaml
+env:
+  - name: ENV1
+    value: foo
+  - name: ENV2
+    value: bar
+  - name: ENV5
```

Comment:
I am not sure that this is a useful feature. It is very complex and I'm not sure why it is necessary, or even why it is an improvement? I can see adding an "append" command.